src/input/type_base.cc

```diff
@@ -80,6 +81,15 @@ void TypeBase::lazy_finish() {
 
 
 
+boost::shared_ptr<TypeBase> TypeBase::substitute_instance_type(boost::shared_ptr<TypeBase> type_ptr) {
```

Comment:
Where this is used? I suppose in lazy_finish. But there we should go trough TypeRepository<Instance> and we know that Instance types are there. So we can call make_instance directly.